analyses/mutational-signatures/08-explore_hypermutators.Rmd

```diff
@@ -25,24 +25,34 @@ library(colorblindr)
 proj_root_path  <- file.path( rprojroot::find_root(rprojroot::has_dir(".git")) )
 analysis_path   <- file.path(proj_root_path, "analyses")
 plot_path       <- file.path(analysis_path, "mutational-signatures", "plots", "cns")
+results_path       <- file.path(analysis_path, "mutational-signatures", "results")
 
 # Input files
 fitted_file    <- file.path(analysis_path, "mutational-signatures", "results", "deconstructsigs_exposures_merged.tsv")
 meta_file      <- file.path(proj_root_path, "data", "pbta-histologies.tsv")
 tmb_all_file    <- file.path(proj_root_path, "data", "pbta-snv-consensus-mutation-tmb-all.tsv")
 tmb_coding_file <- file.path(proj_root_path, "data", "pbta-snv-consensus-mutation-tmb-coding.tsv")
 td_palette_file <- file.path(proj_root_path, "figures", "palettes", "tumor_descriptor_palette.tsv")
+bin_palette_file <- file.path(proj_root_path, "figures", "palettes", "binary_color_palette.tsv")
 
 # Output files
 heatmap_file <- file.path(plot_path, "hypermutator_sigs_heatmap.pdf")
 heatmap_legends_file <- file.path(plot_path, "hypermutator_sigs_heatmap_legends.pdf")
+sig_matrix_file <- file.path(results_path, "hypermutator_sig_matrix.tsv")
 ```
 
 First, we read in data and prepare TMB information.
 
 ```{r read_prep_data, warning=FALSE}
-# Read in tumor descriptor palette
+# Read in tumor descriptor and palettes
 td_palette <- readr::read_tsv(td_palette_file)
+bin_palette <- readr::read_tsv(bin_palette_file) %>%
+  mutate(color_names = case_when(
+        color_names == "binary_1" ~ "Solid Tissue",
+        color_names == "binary_2" ~ "Derived Cell Line",
+        TRUE ~ as.character(color_names)))
```

Comment:
NA can be dropped here and remove line 55. Also no more need for `TRUE` line since there will only be 2 options.

```suggestion
bin_palette <- readr::read_tsv(bin_palette_file) %>%
  filter(color_names != "na_color") %>%
  mutate(color_names = case_when(
        color_names == "binary_1" ~ "Solid Tissue",
        color_names == "binary_2" ~ "Derived Cell Line")
    )
```

Reply:
nice